apps/mobile/src/components/Feed/Socialize/ProcessedText.tsx

```diff
@@ -0,0 +1,255 @@
+import { useNavigation } from '@react-navigation/native';
+import { ReactNode, useCallback, useMemo, useRef } from 'react';
+import { Text, TextProps } from 'react-native';
+import { graphql, useFragment } from 'react-relay';
+
+import { GalleryBottomSheetModalType } from '~/components/GalleryBottomSheet/GalleryBottomSheetModal';
+import { Typography } from '~/components/Typography';
+import { ProcessedTextFragment$key } from '~/generated/ProcessedTextFragment.graphql';
+import { ProcessedTextMentionFragment$key } from '~/generated/ProcessedTextMentionFragment.graphql';
+import { MainTabStackNavigatorProp } from '~/navigation/types';
+import { VALID_URL } from '~/shared/utils/regex';
+
+import { WarningLinkBottomSheet } from '../Posts/WarningLinkBottomSheet';
+
+const MARKDOWN_LINK_REGEX = /\[([^\]]+)]\((https?:\/\/[^\s]+)\)/g;
+
+type LinkProps = {
+  value?: string;
+  url: string;
+};
+
+const LinkComponent = ({ url, value }: LinkProps) => {
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+  const handleLinkPress = useCallback(() => {
+    bottomSheetRef.current?.present();
+  }, []);
+
+  return (
+    <>
+      <Text className="text-shadow" onPress={handleLinkPress}>
+        {value ?? url}
+      </Text>
+      <WarningLinkBottomSheet redirectUrl={url} ref={bottomSheetRef} />
+    </>
+  );
+};
+
+type MentionProps = {
+  mention: string;
+  mentionRef: ProcessedTextMentionFragment$key;
+};
+
+const MentionComponent = ({ mention, mentionRef }: MentionProps) => {
+  const mentionData = useFragment(
+    graphql`
+      fragment ProcessedTextMentionFragment on Mention {
+        __typename
+        entity {
+          __typename
+          ... on GalleryUser {
+            __typename
+            username
+          }
+          ... on Community {
+            __typename
+            contractAddress {
+              __typename
+              address
+              chain
+            }
+          }
+        }
+      }
+    `,
+    mentionRef
+  );
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const onMentionPress = useCallback(() => {
+    if (mentionData.entity?.__typename === 'GalleryUser') {
+      navigation.navigate('Profile', {
+        username: mentionData.entity.username ?? '',
+      });
+      return;
+    }
+
+    if (mentionData.entity?.__typename === 'Community') {
+      navigation.navigate('Community', {
+        contractAddress: mentionData.entity.contractAddress?.address ?? '',
+        chain: mentionData.entity.contractAddress?.chain ?? '',
+      });
+    }
+  }, [mentionData, navigation]);
+
+  return (
+    <Text className="text-shadow" onPress={onMentionPress}>
+      {mention}
+    </Text>
+  );
+};
+
+type CommentProps = {
```

Comment:
lets rename the type too to match the component name if it's no longer just for comments